velox/common/base/Pointers.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::velox {
+
+/// Used to dynamically cast a unique pointer from 'SourceType' to
+/// 'DestinationType'. The raw object is moved from 'srcPtr' to 'dstPtr' on
+/// success. The function throws if the cast fails, and the raw object is also
+/// freed.
+template <typename SourceType, typename DestinationType>
+inline void castUniquePointer(
```

Comment:
What if we move this flush policy specific logic to FlushPolicy.h, and call it setFlushPolicy or similar sort. Do we know if this type of pointer cast will be commonly used so that we want to make generic in common/base?

Reply:
We have seen similar use cases in past but couldn't find out. Have chat with @Yuhta, it might be good to make it a generic utility. Thanks!

Reply:
Then can we add a NOTE: to indicate its performance downside so that it is not mis-used? Thanks